control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -964,86 +870,153 @@ impl CreateVolumeExeVal for CreateVolume {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ReplicaWithLabels {
```

Comment:
Probably all these you added can just be pub(super)
Also missing doc comments

```suggestion
pub(super) struct ReplicaWithLabels {
```